agent/api_test.go

```diff
@@ -351,6 +365,42 @@ func createMockedErrorResponder(t *testing.T, method string) func(req *http.Requ
 	}
 }
 
+func createAuthorizationHeaderChecker(t *testing.T, method string, tokenType authorization.TokenType) func(req *http.Request) *http.Response {
+	return func(req *http.Request) *http.Response {
+		createServerError := func(message string) *http.Response {
+			responseError := `{
+				"error": {
+					"type": "MOCK_SERVER_ERROR",
+					"message": "` + message + `"
+				}
+			}`
+
+			return &http.Response{
+				StatusCode: 400,
+				Body:       ioutil.NopCloser(bytes.NewBufferString(responseError)),
+				Header:     make(http.Header),
+			}
+		}
+		authHeader := req.Header.Get("Authorization")
+
+		if tokenType == authorization.BasicToken && authHeader != "Basic access_token" {
+			t.Errorf("Invalid Authorization header: %s", authHeader)
+			return createServerError("Invalid Authorization")
+		}
+
+		if tokenType == authorization.BearerToken && authHeader != "Bearer access_token" {
+			t.Errorf("Invalid Authorization header: %s", authHeader)
+			return createServerError("Invalid Authorization")
+		}
```

Comment:
you could extract it into functor and pass it as argument to `createAuthorizationHeaderChecker` - `createAuthorizationHeaderChecker` doesn't need to have knowledge of all supported test cases

Reply:
Also pls have a look at `TestAuthorIDHeader` test - it does very similar stuff to what you want to achieve